edb/server/mng_port/edgecon.pyx

```diff
@@ -1663,7 +1681,8 @@ cdef class EdgeConnection:
             # in this situation we just silently exit.
             self.abort()
 
-        except ConnectionAbortedError:
+        except (ConnectionAbortedError, pgerror.BackendQueryCancelledError):
```

Comment:
Postgres might cancel on its own, or it might be external cancellation, should we always `abort()`?  Maybe only if `self._cancelled`?

Reply:
> Postgres might cancel on its own

How and why?

Reply:
Due to a timeout `statement_timeout` or `lock_timeout`, and also on a hot standby node when the async recovery process conflicts with the query.

Reply:
Alright, I've fixed it to only abort when we're cancelling the protocol.